### PR TITLE
feat: Allows placing a domain onto the marketplace contract #14

### LINKED ISSUE #14: List domain: Hooking up controller with RNS Smart contracts
Hooking it up with the smart contracts (dependency smart contracts done) for the List for sale functionality.

**User Story:** [List Domain for Sale](https://github.com/rsksmart/rif-marketplace-project/issues/11)

### PR DESCRIPTION
**Pull request by jurajpiar**
Closes #14.

EDIT: The tutorial has been replaced with https://hackmd.io/CEIhzk6ZTwSOEzLIyToLsQ?view . Keeping the current one for reference.

# Dependencies:
0. NVM
1. Docker
2. the [Dev](https://github.com/rsksmart/rif-marketplace-dev) project for database, db admin and the ganache network
3. the [Rns-Dev-Env](https://github.com/rsksmart/rif-marketplace-rns-dev-env) project to deploy the rns suite to ganache (You'll need to `npm i` in node 10 (nvm use 10))
4. the [Cache](https://github.com/rsksmart/rif-marketplace-cache) project's **feat/rns-entities** branch for rns cache

# Setup _(After installing all required npm modules)_:
1. Run Dev with `docker-compose-up`
2. Copy the first **private key** from the output and add account to your wallet
3. Run the deploy script with `npx truffle deploy --network ganache`
4. Copy the Rif token address from the output and add token to your wallet account
5. Copy Rif token, RSK Owner and Simple Placements addresses from the output of _(3.)_ to .env file in the UI project
6. In the Cache project add _config/local.json_ (you can ignore the addresses. the BC part of the Cache doesn't work. this is just to prevent it from error-ing out):
```
{
    "host": "localhost",
    "port": 3030,
    "blockchain": {
        "provider": "ws://127.0.0.1:4445/websocket"
    },
    "storage": {
        "contractAddress": "0x83C5541A6c8D2dBAD642f385d8d06Ca9B6C731ee",
        "events": [
            "CapacitySet",
            "MaximumDurationSet",
            "PriceSet"
        ],
        "eventsEmitter": {
            "confirmations": 1,
            "startingBlock": "genesis"
        }
    },
    "rns": {
        "contractAddress": "0xd6681FD2d471c044BDdDc9DF00D1Af195Bd58bfa",
        "events": [
            "Transfer",
            "ExpirationChanged"
        ],
        "eventsEmitter": {
            "confirmations": 1,
            "startingBlock": "genesis"
        },
        "newBlockEmitter": {
            "polling": false
        }
    },
    "rns-reverse": {
        "contractAddress": "0x8650B271F7cC565aa604F9f98Aa1A5B723B5b2f3",
        "events": [
            "NameChanged"
        ],
        "eventsEmitter": {
            "confirmations": 1,
            "startingBlock": "genesis"
        },
        "newBlockEmitter": {
            "polling": false
        }
    },
    "db": "postgres://dev:dev@localhost/rif_marketplace_cache",
    "log": {
        "level": "debug",
        "path": "./logs"
    }
}
```
7. comment out lines 15-19 in the src/rns/hooks/domain-offer.hook.ts
8. In the browser navigate to `http://localhost:8080`, where
```
db=rif_marketplace_cache
name=dev
pass=dev
```
9. insert into rns_domain table
```
ownerAddress=your new wallet account address
name=bob.rsk _it is important that you use [bob, alice, charlie] for now as these are def owned by the first account_
```
10. run UI
11. follow the User story [#11 (List RNS Domain for Sale)](https://github.com/rsksmart/rif-marketplace-project/issues/11)

### CODE REVIEW

**Comment by julianmrodri** [comment]
1)  I had to do an extra step between 2 and 3. I had to compile the RNS Suite contracts within the @rsksmart/rns-suite folder (in node_modules) otherwise It couldnt proceed deploying RNS.
So I executed "truffle compile" in   "rif-marketplace-rns-dev-env/node_modules/@rsksmart/rns-suite". Not finished yet with the deployment but identified that already.

2) Another issue is that we have to specify POSTGRESQL in the database connection modal (in addition to including user,pass and db name). But then when I connect I dont see any table RNS_Domain to enter those values you mention.

I couldnt move beyond #2. Also Im not sure when to run the Cache probably between steps 9 and 10? Should I  follow all the steps in the Cache README?

**Comment by jurajpiar** [comment]
Ah yea sorry. I forgot that you have to run the rns-dev-env project in node 10. I'll update the desc.

File: src/components/pages/rns/buy/DomainOffersCheckoutPage.tsx [hunk 1]
```diff
@@ -10,8 +10,24 @@ import { Card, CardContent, CardHeader } from 'rifui/components/atoms/card';
...
+import { MarketListingTypes } from 'models/Market';
+import { Web3Store } from 'rifui/providers/Web3Provider';
+
+import ERC721SimplePlacements from '@rsksmart/rif-marketplace-nfts/build/contracts/ERC721SimplePlacements.json';
+import ERC677 from '@rsksmart/rif-marketplace-nfts/build/contracts/ERC677.json';
+import ERC721 from '@rsksmart/rif-marketplace-nfts/build/contracts/ERC721.json';
 import { colors } from 'rifui/theme';
+const contract = require("@truffle/contract");
+
+function ContractWrapper(artifact, web3, from) {
```
Comment (vojtechsimetka) [inline src/components/pages/rns/buy/DomainOffersCheckoutPage.tsx:23]: this already exists in `utils/blockchain`

File: src/components/pages/rns/sell/DomainCheckoutPage.tsx [hunk 2]
```diff
@@ -96,18 +107,54 @@ const DomainsCheckoutPage = () => {
...
+          isProcessing: true,
+        }
+      })
+
+      const marketPlaceContract = await Contract(ERC721SimplePlacements).at(marketPlaceAddress)
+
+      const rnsContract = await Contract(ERC721).at(rnsAddress)
+
+      const owner = await rnsContract.ownerOf(tokenId)
+      console.log(`ownerOf${name}:${owner}`)
```
Comment (vojtechsimetka) [inline src/components/pages/rns/sell/DomainCheckoutPage.tsx:130]: Please remove all the `console.log`s

**Pull request merged**